Assignment_4.py

```diff
@@ -0,0 +1,318 @@
+            f_path.append("<a href=" + filename + ">" + filename + "</a>")
+
+        # Calculating p-value and t-score
+        if res_type == "Boolean":
+            predictor = sm.add_constant(inp_data[col])
+            logit = sm.Logit(inp_data["target"], predictor)
+            logit_fitted = logit.fit()
+            # Get the stats
+            t_value = round(logit_fitted.tvalues[1], 6)
+            p_value = "{:.6e}".format(logit_fitted.pvalues[1])
+            t_val.append(t_value)
+            p_val.append(p_value)
+            filename = "~/plots/ranking_" + col_name + ".html"
+            m_plot.append("<a href=" + filename + ">" + filename + "</a>")
+            # Plot the figure
+            fig = px.scatter(x=inp_data[col], y=inp_data["target"], trendline="ols")
+            fig.update_layout(
+                title=f"Variable: {col}: (t-value={t_value}) (p-value={p_value})",
+                xaxis_title=f"Variable: {col}",
+                yaxis_title="y",
+            )
+            # fig.show()
+            fig.write_html(
+                file=f"~/plots/ranking_{col_name}.html", include_plotlyjs="cdn"
+            )
+        else:
+            predictor = sm.add_constant(inp_data[col])
+            linear_regression_model = sm.OLS(inp_data["target"], predictor)
+            linear_regression_model_fitted = linear_regression_model.fit()
+
+            # Get the stats
+            t_value = round(linear_regression_model_fitted.tvalues[1], 6)
+            p_value = "{:.6e}".format(linear_regression_model_fitted.pvalues[1])
+            t_val.append(t_value)
+            p_val.append(p_value)
+            # Plot the figure
+            fig = px.scatter(x=inp_data[col], y=inp_data["target"], trendline="ols")
+            fig.update_layout(
+                title=f"Variable: {col}: (t-value={t_value}) (p-value={p_value})",
+                xaxis_title=f"Variable: {col}",
+                yaxis_title="y",
+            )
+            # fig.show()
+            fig.write_html(
+                file=f"{file_path}/ranking_{col_name}.html", include_plotlyjs="cdn"
+            )
+            filename = "~/plots/ranking_" + col_name + ".html"
+            m_plot.append("<a href=" + filename + ">" + filename + "</a>")
+        # Mean Squared Difference
+        if is_pred_categorical(X[col]):
+            pop_prop_1 = inp_data.target.sum() / len(inp_data)
+            uw, w = calculate_mean_of_response_cat(inp_data, col, pop_prop_1)
+        else:
+            pop_prop_1 = inp_data.target.sum() / len(inp_data)
+            uw, w = calculate_mean_of_response(inp_data, col, pop_prop_1)
+        msd_uw.append(uw)
+        msd_w.append(w)
+
+    output_df["Cat/Con"] = out
+    output_df["Plot_Link"] = f_path
+    output_df["t-value"] = t_val
+    output_df["p-value"] = p_val
+    output_df["m-plot"] = m_plot
+    output_df["RF-FI"] = importance
+    output_df["MeanSqDiff"] = msd_uw
+    output_df["MeanSqDiffWeighted"] = msd_w
```

Comment:
This will include the difference with mean of response (weighted and unweighted), but I am not sure if you have created a plot representing the same. I think that was a part of this assignment.

Reply:
Oh completely skipped that.. Thanks a ton.. will add the plot

Reply:
Once you are done with your code..lets run for same csv and then we can verify these outputs to see if it matches

Reply:
Added the plot